modules/formulize/class/durationElement.php

```diff
@@ -0,0 +1,271 @@
+					}
+				}
+			}
+			return new XoopsFormLabel($caption, trim($output), $markupName);
+		}
+
+		// Create container for inputs
+		$container = new XoopsFormElementTray($caption, ' ');
+
+		// Add input field for each enabled time unit
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			$unitMarkupName = $markupName . '[' . $unit . ']';
+			if ($ele_value['show_' . $unit]) {
+				$value = isset($ele_value['values'][$unit]) ? $ele_value['values'][$unit] : '';
+				${"input_$unit"} = new XoopsFormText(
+					ucfirst($unit).":",
+					$unitMarkupName,
+					$ele_value['size'],
+					5,
+					$value,
+					false,
+					true
+				);
+				${"input_$unit"}->setExtra("min='0'");
+				${"input_$unit"}->setExtra("class='formulize-duration-element-input numbers-only-textbox'");
+				${"input_$unit"}->setExtra(" onchange=\"javascript:formulizechanged=1;\" jquerytag=\"$unitMarkupName\" ");
+				$container->addElement(${"input_$unit"});
+			}
+		}
+
+		$renderedElement = $container->render();
+
+		$form_ele = new XoopsFormLabel(
+			$caption,
+			trans($renderedElement),
+			$markupName
+		);
+
+		return $form_ele;
+	}
+
+	// Validate the duration is within bounds
+	function generateValidationCode($caption, $markupName, $element, $entry_id = false)
+	{
+		$validationCode = array();
+		$ele_value = $element->getVar('ele_value');
+
+		// Basic required validation if element is required
+		if ($element->getVar('ele_req')) {
+			$validationCode[] = "var hasValue = false;\n";
+			foreach ($this->timeUnits as $unit => $multiplier) {
+				if ($ele_value['show_' . $unit]) {
+					$validationCode[] = "if(myform['{$markupName}[{$unit}]'].value != '') hasValue = true;\n";
+				}
+			}
+			$validationCode[] = "if(!hasValue) {\n";
+			$validationCode[] = "    window.alert('Please enter a duration.');\n";
+			$validationCode[] = "    return false;\n";
+			$validationCode[] = "}\n";
+		}
+
+		// Validate min/max duration
+		$validationCode[] = "var totalMinutes = 0;\n";
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			if ($ele_value['show_' . $unit]) {
+				$validationCode[] = "if(myform['{$markupName}[{$unit}]'].value != '') {\n";
+				$validationCode[] = "    totalMinutes += parseInt(myform['{$markupName}[{$unit}]'].value) * $multiplier;\n";
+				$validationCode[] = "}\n";
+			}
+		}
+
+		if ($ele_value['min_minutes'] > 0) {
+			$validationCode[] = "if(totalMinutes < {$ele_value['min_minutes']}) {\n";
+			$validationCode[] = "    window.alert('Duration must be at least {$ele_value['min_minutes']} minutes.');\n";
+			$validationCode[] = "    return false;\n";
+			$validationCode[] = "}\n";
+		}
+
+		if ($ele_value['max_minutes'] > 0) {
+			$validationCode[] = "if(totalMinutes > {$ele_value['max_minutes']}) {\n";
+			$validationCode[] = "    window.alert('Duration must not exceed {$ele_value['max_minutes']} minutes.');\n";
+			$validationCode[] = "    return false;\n";
+			$validationCode[] = "}\n";
+		}
+
+		return $validationCode;
+	}
+
```

Comment:
I think you need to have the methods that are not included here, and just have them return $value or true or whatever, so that when those parts of the system kick in, you don't get a fatal error because the method is not found. I'm thinking primarily of the formatUserInputForDatabase or whatever it's called, that is used for searches, to translate user readable values into DB values. For example, if we wanted to support 1:30 as a way of representing 90 seconds, then we would have to translate that into seconds, because the search in the DB has to be done with seconds. As it stands just typing in total seconds into the search box should work (when the right method is present, even if it's doing nothing). But I wonder if we need to do more here... if the duration is 2 hours, people will want to search for 2, not 7200. (We are storing in total seconds, right?). But what search convention should this have? Maybe H:M:S ? There is a cheat sheet that tells people what all the search tricks are, this should definitely go in there. You can find the sheet by clicking the question mark on the far left of the search box row on a standard list.

Reply:
Sorry, I'm talking about seconds up there, and it's days, hours, minutes we're talking about. But anyway, same idea. Though days really throws a wrench in the X:X:X format. Hmmmm. This is tricky.

Reply:
What might be useful as a first pass is allowing the user to enter a single value with the suffix. E.g. `1h` This will allow them to do useful searches like `> 1h` or `< 30m`

Reply:
Amazing!